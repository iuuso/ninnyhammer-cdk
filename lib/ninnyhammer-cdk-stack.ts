import * as s3 from '@aws-cdk/aws-s3';
import * as cloudfront from '@aws-cdk/aws-cloudfront';
import * as origins from '@aws-cdk/aws-cloudfront-origins';
import { CfnOutput, Construct, Stage, Stack, StageProps, StackProps } from '@aws-cdk/core';
import { SecurityPolicyProtocol } from '@aws-cdk/aws-cloudfront';

export class WebLayerStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
    
    const webbucket = new s3.Bucket(this, 'NinnyhammerBucket', {
      bucketName: 'ninnyhammer-web-bucket',
    });

    const cfDistribution = new cloudfront.Distribution(this, 'NinnyhammerCF', {
      defaultBehavior: {
        origin: new origins.S3Origin(webbucket)
      },
      minimumProtocolVersion: SecurityPolicyProtocol.TLS_V1_2_2019,
      domainNames: ['juusokarlstrom.com'],
    });
};
}