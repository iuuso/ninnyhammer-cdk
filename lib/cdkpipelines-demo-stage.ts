import { CfnOutput, Construct, Stage, StageProps } from '@aws-cdk/core';
// import { CdkpipelinesDemoStack } from './cdkpipelines-demo-stack';
import { NinnyhammerCdkStack} from './ninnyhammer-cdk-stack';

/**
 * Deployable unit of web service app
 */
export class CdkpipelinesDemoStage extends Stage {
  public readonly urlOutput: CfnOutput;
  
  constructor(scope: Construct, id: string, props?: StageProps) {
    super(scope, id, props);

    // const service = new CdkpipelinesDemoStack(this, 'WebService');
    const service = new NinnyhammerCdkStack(this, "WebService");
    
    // Expose CdkpipelinesDemoStack's output one level higher
    this.urlOutput = service.urlOutput;
  }
}
