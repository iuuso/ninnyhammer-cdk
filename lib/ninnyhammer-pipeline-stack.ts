import * as codepipeline from '@aws-cdk/aws-codepipeline';
import * as codepipeline_actions from '@aws-cdk/aws-codepipeline-actions';
import { Construct, SecretValue, Stack, StackProps } from '@aws-cdk/core';
import { CdkPipeline, SimpleSynthAction } from "@aws-cdk/pipelines";
import { env } from 'node:process';

import { NinnyhammerWebLayer } from './ninnyhammer-cdk-stack';

export class NinnyhammerPipelineStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const sourceArtifact = new codepipeline.Artifact();
    const cloudAssemblyArtifact = new codepipeline.Artifact();
 
    const pipeline = new CdkPipeline(this, 'Pipeline', {
      // The pipeline name
      pipelineName: 'Ninnyhammer',
      cloudAssemblyArtifact,

      // Where the source can be found
      sourceAction: new codepipeline_actions.GitHubSourceAction({
        actionName: 'GitHub',
        output: sourceArtifact,
        oauthToken: SecretValue.secretsManager('github-token'),
        owner: 'iuuso',
        repo: 'ninnyhammer-cdk',
      }),

       // How it will be built and synthesized
       synthAction: SimpleSynthAction.standardNpmSynth({
         sourceArtifact,
         cloudAssemblyArtifact,
         
         // We need a build step to compile the TypeScript Lambda
         installCommand: 'npm install -g aws-cdk && npm i',
         buildCommand: 'npm install && npm run build'
       }),
    });

    // This is where we add the application stages
    pipeline.addApplicationStage(new NinnyhammerWebLayer(this, 'Prod', {
      env: {
        account: process.env.CDK_DEFAULT_ACCOUNT,
        region: process.env.CDK_DEFAULT_REGION,
      }
    }))
  }
}
