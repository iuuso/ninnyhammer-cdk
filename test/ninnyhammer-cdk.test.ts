import { expect as expectCDK, matchTemplate, MatchStyle } from '@aws-cdk/assert';
import * as cdk from '@aws-cdk/core';
import * as NinnyhammerCdk from '../lib/ninnyhammer-cdk-stack';

// test('Empty Stack', () => {
//     const app = new cdk.App();
//     // WHEN
//     const stack = new NinnyhammerCdk.NinnyhammerPipelineStack(app, 'MyTestStack');
//     // THEN
//     expectCDK(stack).to(matchTemplate({
//       "Resources": {}
//     }, MatchStyle.EXACT))
// });
