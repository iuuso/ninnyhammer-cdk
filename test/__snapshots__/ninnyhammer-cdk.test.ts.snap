// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bucket with the name ninnyhammer-bucket 1`] = `
Object {
  "Resources": Object {
    "Bucket83908E77": Object {
      "DeletionPolicy": "Retain",
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
